pkg/antctl/raw/supportbundle/command_test.go

```diff
@@ -0,0 +1,398 @@
+			antreaClientset: fakeclientset.NewSimpleClientset(agentInfo1, agentInfo2),
+			expectedClients: []string{"node-3"},
+		},
+		{
+			name:            "no-ops due to error in listing antreaagentinfos",
+			antreaClientset: fakeclientset.NewSimpleClientset(),
+			prepareReactor: func(antreaClientset *fakeclientset.Clientset, k8sClientset *fake.Clientset) {
+				antreaClientset.PrependReactor("list", "antreaagentinfos", func(action cgtesting.Action) (handled bool, ret runtime.Object, err error) {
+					return true, &v1beta1.AntreaAgentInfoList{}, errors.New("error listing antreaagentinfos")
+				})
+			},
+			expectedErr: "error listing antreaagentinfos",
+		},
+		{
+			name:            "no-ops due to error in listing nodes",
+			antreaClientset: fakeclientset.NewSimpleClientset(),
+			k8sClientset:    fake.NewSimpleClientset(),
+			prepareReactor: func(antreaClientset *fakeclientset.Clientset, k8sClientset *fake.Clientset) {
+				k8sClientset.PrependReactor("list", "nodes", func(action cgtesting.Action) (handled bool, ret runtime.Object, err error) {
+					return true, &v1.NodeList{}, errors.New("error listing nodes")
+				})
+			},
+			expectedErr: "error listing nodes",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.prepareReactor != nil {
+				tt.prepareReactor(tt.antreaClientset, tt.k8sClientset)
+			}
+			clients, err := createAgentClients(tt.k8sClientset, tt.antreaClientset, clientConfig, "", nameList)
+			if tt.expectedErr != "" {
+				assert.ErrorContains(t, err, tt.expectedErr)
+			} else {
+				require.NoError(t, err)
+				agentClients := []string{}
+				for client := range clients {
+					agentClients = append(agentClients, client)
+				}
+				sort.Strings(agentClients)
+				assert.ElementsMatch(t, tt.expectedClients, agentClients)
+			}
+		})
+	}
+}
+
+func TestRequest(t *testing.T) {
+	agentClients := map[string]rest.Interface{}
+	supportBundle := &systemv1beta1.SupportBundle{
+		ObjectMeta: metav1.ObjectMeta{Name: "agent"},
+		Status:     systemv1beta1.SupportBundleStatusCollected,
+	}
+	resp, err := json.Marshal(supportBundle)
+	require.NoError(t, err)
+	for _, name := range nameList {
+		restClient := &fakerest.RESTClient{
+			NegotiatedSerializer: scheme.Codecs,
+			GroupVersion:         systemv1beta1.SchemeGroupVersion,
+			Client: fakerest.CreateHTTPClient(func(request *http.Request) (*http.Response, error) {
+				resp := &http.Response{
+					StatusCode: http.StatusOK,
+					Body:       io.NopCloser(bytes.NewBuffer(resp)),
+				}
+				return resp, nil
+			}),
+		}
+		agentClients[name] = restClient
+	}
+	controllerClient := &fakerest.RESTClient{
+		NegotiatedSerializer: scheme.Codecs,
+		GroupVersion:         systemv1beta1.SchemeGroupVersion,
+		Client: fakerest.CreateHTTPClient(func(request *http.Request) (*http.Response, error) {
+			resp := &http.Response{
+				StatusCode: http.StatusOK,
+				Body:       io.NopCloser(bytes.NewBuffer(resp)),
```

Comment:
nit: it's a bit surprising than you use a supportbundle with name "agent" for the controller.